pkg/controller/installconfig/installconfig_controller.go

```diff
@@ -81,27 +75,33 @@ func (r *ReconcileInstallConfig) Reconcile(request reconcile.Request) (reconcile
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling InstallConfig")
 
-	// Fetch the InstallConfig instance
 	installConfig := &v1alpha1.InstallConfig{}
 	err := r.client.Get(context.TODO(), request.NamespacedName, installConfig)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
 			return reconcile.Result{}, nil
 		}
-		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
 
+	var errors []error
 	if isResourceCreated, err := r.EnsureCheSubscription(reqLogger, installConfig); err != nil {
-		return reconcile.Result{}, err
+		errors = append(errors, err)
+	} else if isResourceCreated {
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	if isResourceCreated, err := r.EnsureTektonSubscription(reqLogger, installConfig); err != nil {
```

Comment:
why don't we need to have watch added for che_sub and tekton_sub .. in case they deleted or modified .. we need to run this `Reconcile()` and do needful ??

Reply:
With current implementation, we are just creating ns/og/subscription once, after that if user deletes anyone of them, operator is not going to maintain it's state. This way if someone wants to uninstall any specific operator from che/tekton they can do it. Otherwise, end user won't be able to remove any specific operator. Also if we decided later to support this usecase, we can watch subscription as secondary resources